### ISSUE TITLE
feature request: fuzzy lint for idents implementing io.Closer interface

### ISSUE THREAD

**Issue by jzacsh** [golang/lint#220]
I'd like golint to complain about the second line here:
```go
func Foo(string path) {
  id, err := os.Open(path)

  ....  // linty code.. `id.Close()` never called here (defer or otherwise)
}
```
**tl;dr** I'm guessing a heuristic like the following can catch this common case:
 1. `id` was assigned to in the current scope
 2. `id` has a `.Close()` method (ie: implements the [`io.Closer` interface](https://golang.org/pkg/io/#Closer))
 3. `id.Close()` not called in the top level of this scope

By "in the top level" of bullet 3 I'm speaking from not having thoroughly read https://github.com/golang/lint/blob/master/lint.go - but I'm guessing it is probably a quicker hack to implement this for a surface check _(between the open and closing brace of `Foo`, and not in the context of any child scopes)_. If this is the case, we can at least drop the confidence of the lint if we see `id` passed anywhere:
```go
func Foo(string path) {
  id, err := os.Open(path)
  doStuffWithFile(&id)
  return // can't easily be sure id.Close() wasn't called... we lower lint confidence
}
```

**Comment by dsnet** [comment]
I don't believe this belongs in `golint`. This is clearly an issue of correctness and the same issue should be filed in https://github.com/golang/go/issues in regards to `cmd/vet`, who's job to check for correctness.
